src/VisualStudio/Core/Def/Microsoft.VisualStudio.LanguageServices.csproj

```diff
@@ -201,6 +201,7 @@
     <PackageReference Include="NuGet.VisualStudio" Version="$(NuGetVisualStudioVersion)" />
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.SDK.EmbedInteropTypes" Version="$(MicrosoftVisualStudioSDKEmbedInteropTypesVersion)" />
+    <PackageReference Include="System.Threading.Tasks.Dataflow" Version="$(SystemThreadingTasksDataflowVersion)" />
```

Comment:
why?

Reply:
Moving to newer Shell packages seems to cause MSBuild version conflicts from implicitly chosen version and version coming from Shell packages. Few more are still there: https://dev.azure.com/dnceng/public/_build/results?buildId=367711&view=logs

```
Build FAILED.

F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\EditorFeatures\TestUtilities\Roslyn.Services.Test.Utilities.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\EditorFeatures\TestUtilities\Roslyn.Services.Test.Utilities.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\Test\Diagnostics\Roslyn.Hosting.Diagnostics.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\EditorFeatures\CSharpTest2\Microsoft.CodeAnalysis.CSharp.EditorFeatures2.UnitTests.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\EditorFeatures\CSharpTest\Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\EditorFeatures\CSharpTest\Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\Tools\IdeBenchmarks\IdeBenchmarks.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\Tools\IdeBenchmarks\IdeBenchmarks.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\EditorFeatures\Test\Microsoft.CodeAnalysis.EditorFeatures.UnitTests.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\Interactive\HostTest\InteractiveHost.UnitTests.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\EditorFeatures\VisualBasicTest\Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests.vbproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\EditorFeatures\VisualBasicTest\Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests.vbproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\Features\LanguageServer\ProtocolUnitTests\Microsoft.CodeAnalysis.LanguageServer.Protocol.UnitTests.csproj]
F:\workspace.2\_work\1\s\.tools\msbuild\16.0.0-alpha\tools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2114,5): error MSB3277: Found conflicts between different versions of "System.Threading.Tasks.Dataflow" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [F:\workspace.2\_work\1\s\src\Features\LanguageServer\ProtocolUnitTests\Microsoft.CodeAnalysis.LanguageServer.Protocol.UnitTests.csproj]
```

Reply:
@dotnet/roslyn-infrastructure for guidance. There is probably a better way to resovle these conflicts.

Reply:
Hmm.. We've been seeing this issue when validating our 16.3 NuGet packages on .NET Framework projects. In that case adding the "Microsoft.VisualStudio.LanguageServices" package to the test project adds the 4.5.24 version of the Tasks.Dataflow package whereas VS seems to be building against 4.6.3 and build warnings are given.

Reply:
>  adds the 4.5.24 version of the Tasks.Dataflow package whereas VS seems to be building against 4.6.3 and build warnings are given.

Those are the exact conflicts I have been seeing. If you just try to apply the changes to `Versions.props` in this PR on top of master, you will see these MSBuild warnings locally on building Microsoft.VisualStudio.LanguageServices

Reply:
I've reached out to get some guidance on this. For the time being I think this is ok and hopefully solves both of our issues.

Reply:
Thanks. Note that I would need to add this package reference to all the test projects mentioned in https://github.com/dotnet/roslyn/pull/38894#discussion_r328855499 as well.